osquery/config/config.cpp

```diff
@@ -530,6 +541,54 @@ void stripConfigComments(std::string& json) {
   json = sink;
 }
 
+std::map<std::string, std::string> Config::restoreConfigBackup() {
+  if (!FLAGS_config_enable_backup) {
+    return std::map<std::string, std::string>();
+  }
+
+  VLOG(1) << "Restoring backed up config from the database";
+  std::vector<std::string> keys;
+  std::map<std::string, std::string> config;
+
+  WriteLock lock(config_backup_mutex_);
+  scanDatabaseKeys(kPersistentSettings, keys, kConfigPersistencePrefix);
+
+  for (const auto& key : keys) {
+    std::string value;
+    Status status = getDatabaseValue(kPersistentSettings, key, value);
+    if (!status.ok()) {
+      continue;
+    }
+    config[key.substr(kConfigPersistencePrefix.length())] = std::move(value);
+  }
+
+  return config;
+}
+
+void Config::backupConfig(const std::map<std::string, std::string>& config) {
```

Comment:
Why this function is an member of class `Config`?

Reply:
Nice. There are quite a lot of classes in this file. I want to somehow protect it from other classes calling it. Will add static(as it's not changing the class state) if you don't have better suggestion.

Reply:
UPD UPD: @akindyakov 
It is using kConfigPersistencePrefix. Static also makes less sense as this class is singleton anyways.

Do you have any improvement in mind? Would love to hear.